### PR TITLE
Implement enumumerate() &  timestamps()

### PR DESCRIPTION
**Pull request by DonBraulio**

### CODE REVIEW

File: temporian/core/operators/timestamps.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    Datetime timestamps are converted to unix timestamps.
+
+    Integer timestamps example:
+        ```python
+        >>> from datetime import datetime
+        >>> evset = tp.event_set(
+        ...    timestamps=[1, 2, 3, 5],
+        ...    name='simple_timestamps'
+        ... )
+        >>> tp.timestamps(evset.node()).run(evset)
```
Comment (achoum) [inline temporian/core/operators/timestamps.py:71]: I like those examples.

Would it make sense to also create an example with an operation on the timestamps.

For example:

tp.timestamps(...) * 2
or  tp.timestamps(...) <= datetime("2020-01-02").timestamp()

File: temporian/implementation/numpy/operators/enumerate.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        output_schema = self.output_schema("output")
+
+        # Create output EventSet
+        output_evset = EventSet(data={}, schema=output_schema)
+
+        # Fill output EventSet's data
+        for index_key, index_data in input.data.items():
+            output_evset[index_key] = IndexData(
+                [np.indices(index_data.timestamps.shape).flatten()],
```
Comment (achoum) [inline temporian/implementation/numpy/operators/enumerate.py:44]: Would using numpy.arange be easier ?

https://numpy.org/doc/stable/reference/generated/numpy.arange.html

Nit: dtype=np.int64

**Review by achoum — COMMENTED** [review]
Awesome :)

File: temporian/core/operators/timestamps.py (hunk 1 above)
Comment (DonBraulio) [reply temporian/core/operators/timestamps.py:71]: I like it! I think the comparison is very useful.

**Comment by DonBraulio** [comment]
Thanks! All suggestions implemented.

**Review by achoum — APPROVED** [review]
Awesome.
Looking forward to use those ops :).

Make sure to allign with Ian before submitting as he is working on a code-wide pr.

**Comment by ianspektor** [comment]
Make sure to merge main and add @compile decorator to new ops before merging 👮🏼‍♂️

**Pull request merged**